islandora_ocr.install

```diff
@@ -49,115 +50,16 @@ function islandora_ocr_cleanup_derivatives() {
-  // this is in an update hook and the site therefore should not be being messed
-  // with at this time.
-  $slice = islandora_ocr_update_get_generate_ocr_items($conn, $sandbox['current']);
-  if (empty($slice)) {
-    $t = get_t();
-    $sandbox['finished'] = 1;
-    return $t('Updated @count items with generate_ocr flags.', array(
-      '@count' => $sandbox['current'],
-    ));
-  }
-  // Loop through e'm.
-  foreach ($slice as $item) {
-    $sandbox['current']++;
-    $object = islandora_object_load($item['pid']['value']);
-    if ($object) {
-      // Get the generate_ocr value and use it to set the generate_hocr value.
-      $current = $object->relationships->get(ISLANDORA_RELS_EXT_URI, 'generate_ocr', NULL, RELS_TYPE_PLAIN_LITERAL);
-      if (!empty($current)) {
-        $object->relationships->add(ISLANDORA_RELS_EXT_URI, 'generate_hocr', $current[0]['object']['value'], RELS_TYPE_PLAIN_LITERAL);
-      }
-      else {
-        watchdog('islandora_ocr',
-          'Failed to get generate_ocr flag value from @object',
-          array('@object' => $object->id),
-          WATCHDOG_WARNING);
-      }
-    }
-    else {
-      watchdog('islandora_ocr',
-        'Failed to load @object',
-        array('@object' => $item['pid']['value']),
-        WATCHDOG_WARNING);
-    }
-  }
-  // Determine if we're finished.
-  $sandbox['finished'] = (int) floor($sandbox['current'] / $sandbox['total']);
-  $t = get_t();
-  if ($sandbox['finished']) {
-    return $t('Updated @count items with generate_ocr flags.', array(
-      '@count' => $sandbox['total'],
-    ));
-  }
-  else {
-    return $t('@current/@count items updated.', array(
-      '@current' => $sandbox['current'],
-      '@count' => $sandbox['total'],
-    ));
-  }
-}
-
-/**
- * Gets the number of items in the repository with 'generate_ocr' flags.
- *
- * @param IslandoraTuque $conn
- *   Connection to Fedora.
- *
- * @return int
- *   The number of generate_ocr flags in the repo.
- */
-function islandora_ocr_update_get_generate_ocr_flag_count(IslandoraTuque $conn) {
-  $query = <<<EOQ
-SELECT ?pid
-FROM <#ri>
-WHERE {
-  ?pid <http://islandora.ca/ontology/relsext#generate_ocr> ?value .
-}
-EOQ;
-  return $conn->repository->ri->countQuery($query, 'sparql');
-}
-
-/**
- * Gets a chunk of up to 100 items with the generate_ocr flag.
- *
- * @param IslandoraTuque $conn
- *   Connection to Fedora.
- * @param int $offset
- *   Slicing parameter to get the offset.
- *
- * @return array
- *   An array of items to be updated.
- */
-function islandora_ocr_update_get_generate_ocr_items(IslandoraTuque $conn, $offset) {
-  $query = <<<EOQ
-SELECT ?pid
-FROM <#ri>
-WHERE {
-  ?pid <http://islandora.ca/ontology/relsext#generate_ocr> ?value .
-}
-OFFSET $offset
-EOQ;
-  return $conn->repository->ri->sparqlQuery($query, 100);
+  $timestamp = date_iso8601(time());
+  variable_set(ISLANDORA_OCR_UPDATE_7001_TIMESTAMP, $timestamp);
```

Comment:
I'm... almost inclined to suggested making sure there is anything present that this will effect, since the relationship is removed after generation... It only persists when generation is being prevented.